### PR TITLE
Add projects

### PR DESCRIPTION
**Pull request by roalcantara**

### CODE REVIEW

File: spec/models/certification_process_spec.rb [hunk 1]
```diff
@@ -17,4 +17,14 @@
...
     it { is_expected.to delegate_method(:name).to(:segment).with_prefix true }
   end
+
+  describe "scopes" do
+    describe ".enabled" do
+      before { create_list :certification_process, 3 }
+
+      it "finds all the projects of the given user" do
+        expect(CertificationProcess.enabled.collect(&:enabled)).to all be_truthy
+      end
```
Comment (ddomingues) [inline spec/models/certification_process_spec.rb]: This test is not so clear for me. How about to create the list with `enable: true` and assert that you recovery the list with the correct items?

File: app/models/project.rb [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
-  validates_presence_of :user
+  validates :user, presence: true
   validates :segment, presence: true, uniqueness: { scope: :user_id }
+
+  scope :of, ->(user) { where(user: user) }
+  scope :by_segment_name_like, ->(segment_name) {
+    joins(:segment).where("lower(segments.name) like lower(?)", "%#{segment_name}%")
+  }
+  scope :by_segment, ->(segment) { where(segment: segment) }
+  scope :by, lambda { |params|
```
Comment (ddomingues) [inline app/models/project.rb]: humm....  I dunno if it is a good idea...

File: app/models/project.rb [hunk 3]
```diff
@@ -20,4 +27,17 @@ class Project < ApplicationRecord
       all
     end
   }
+
+  def progress=(value)
+    case value
+    when 0
+      self[:status] = :pending unless pending?
```
Comment (ddomingues) [inline app/models/project.rb]: I didn't understand why the `unless`

File: db/migrate/20161029211510_add_type_and_project_task_to_requirements.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class AddTypeAndProjectTaskToRequirements < ActiveRecord::Migration[5.0]
+  def change
+    add_column :requirements, :type, :string, null: false
```
Comment (ddomingues) [inline db/migrate/20161029211510_add_type_and_project_task_to_requirements.rb]: You forgot an index here 😉

File: spec/models/project_spec.rb [hunk 5]
```diff
@@ -52,4 +52,39 @@
       end
     end
   end
+
+  describe "#find_or_initialize_project_tasks" do
```
Comment (roalcantara) [inline spec/models/project_spec.rb]: Do you mean, something like this?
```
it "creates a project task with the given task" do
  project_task = project.project_tasks.create!(task: task)

  expect(project_task.task).to eq task
end
```
🤔

File: spec/support/shared_examples/paranoia_examples.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+RSpec.shared_examples "a Paranoid model" do
+  it { is_expected.to have_db_column :deleted_at }
+  it { is_expected.to have_db_index :deleted_at }
+
+  it "adds a deleted_at where clause" do
+    expect(described_class.all.where_sql).to include('"deleted_at" IS NULL')
+  end
+
+  it "skips adding the deleted_at where clause when unscoped" do
+    expect(described_class.unscoped.where_sql.to_s).not_to include("deleted_at") # to_s to handle nil.
```
Comment (roalcantara) [inline spec/support/shared_examples/paranoia_examples.rb]: I did add the `it { is_expected.to act_as_paranoid }` test, tho. 😁

File: app/controllers/admin/dynamic_forms_controller.rb [hunk 7]
```diff
@@ -29,7 +29,8 @@ def update
 
   def destroy
     @dynamic_form.destroy
-    respond_with :admin, @dynamic_form
+    redirect_to admin_dynamic_forms_path,
+                flash: { info: t("flash.actions.destroy.notice", resource_name: DynamicForm.model_name.human) }
```
Comment (roalcantara) [inline app/controllers/admin/dynamic_forms_controller.rb]: Ya. It supposed to. The thing is, after start using the paranoid strategy in the DynamicForm model, the flash message stopped being displayed. And I couldn't figure out a better way of doing this automatically. Is there?

File: app/models/project_task_dynamic_form_task.rb [hunk 8]
```diff
@@ -8,7 +8,7 @@ class ProjectTaskDynamicFormTask < ProjectTask
   def task=(task)
     super(task)
     if task.instance_of? DynamicFormTask
-      dynamic_form_version = task.dynamic_form.version
+      self.dynamic_form_version = task.dynamic_form.version
```
Comment (roalcantara) [inline app/models/project_task_dynamic_form_task.rb]: Yeah. So, what would be the best approach to fix this offense, then? 🤔

```
➜  equalize git:(add-projects) rubocop -D
...
app/models/project_task_dynamic_form_task.rb:13:7: W: Lint/UselessAssignment: Useless assignment to variable - dynamic_form_version.
      dynamic_form_version = task.dynamic_form.version
```

File: spec/helpers/application_helper_spec.rb [hunk 9]
```diff
@@ -252,4 +252,35 @@ def has_not_updated_by?(track)
       end
     end
   end
+
+  describe "#partial_form_of" do
+    context "when there is a namespace" do
+      context "with one level" do
+        let(:form) { double("form", object: TextField.new) }
+        it "returns the correct partial form according to the given resource" do
```
Comment (roalcantara) [inline spec/helpers/application_helper_spec.rb]: 😂

File: app/controllers/requirements_controller.rb [hunk 10]
```diff
@@ -1,82 +1,73 @@
...
   end
 
-  def set_dynamic_form
-    @dynamic_form = if params[:dynamic_form_id].present?
-                      DynamicForm.find(params[:dynamic_form_id])
-                    else
-                      @requirement.dynamic_form
-                    end
+  def set_project_task
+    @project_task = ProjectTask.find(params[:task_id])
```
Comment (roalcantara) [inline app/controllers/requirements_controller.rb]: Oh! I didn't know how to do it..
Now I get it! 
Cheers! 😂

**Pull request merged**